apis/python/src/tiledbsoma/indexed_dataframe.py

```diff
@@ -200,18 +199,57 @@ def read(
                 query_condition=query_condition,
             )
 
+            # Acceptable ways to index:
+            # * None
+            # * A sequence of coordinates is accepted, one per dimension.
+            # * Sequence length must be at least one and <= number of dimensions.
+            # * If the sequence contains missing coordinates (length less than number of dimensions),
+            #   then "slice(None)" is assumed for the missing dimensions.
+            # * Per-dimension, explicitly specified coordinates can be one of: None, a value, a
+            #   list/ndarray/paarray/etc of values, a slice, etc.
+
             if ids is not None:
-                dim_name = A.schema.domain.dim(0).name
-                if isinstance(ids, list):
-                    sr.set_dim_points(dim_name, ids)
-                elif isinstance(ids, pa.ChunkedArray):
-                    sr.set_dim_points(dim_name, ids)
-                elif isinstance(ids, pa.Array):
-                    sr.set_dim_points(dim_name, pa.chunked_array(ids))
-                elif isinstance(ids, slice):
-                    lo_hi = util.slice_to_range(ids)
-                    if lo_hi is not None:
-                        sr.set_dim_ranges(dim_name, lo_hi)
+                if not isinstance(ids, (list, tuple)):
+                    raise TypeError(
+                        f"ids type {type(ids)} unsupported; expected list or tuple"
+                    )
+                if len(ids) < 1 or len(ids) > A.schema.domain.ndim:
+                    raise ValueError(
+                        f"ids {ids} must have length between 1 and ndim ({A.schema.domain.ndim}); got {len(ids)}"
+                    )
+
+                for i, dim_ids in enumerate(ids):
+                    # Example: ids = [None, 3, slice(4,5)]
+                    # dim_ids takes on values None, 3, and slice(4,5) in this loop body.
+                    dim_name = A.schema.domain.dim(i).name
+                    if dim_ids is None:
+                        pass  # No constraint; select all in this dimension
+                    elif isinstance(dim_ids, int):
+                        # TO DO: Support non-int index types when we have non-int index support
```

Comment:
If this is a very temporary work-around, then it is OK as is.  But if longer term, then...the spec pretty clearly says that an error should be raised in `create` if the schema is unsupported.  Non-int index types would fall into that category.

In this case (not a short-term work-around), I recommend you add this check (for int) to `create` - ie, if we don't currently support non-int-indexed dataframes, then it should fail _immediately_ in create, not wait until a read or write.

Reply:
Temporary check added in `create()` pending #418 and #419 which have issues dedicated to them